crates/bencode/src/bencode_value.rs

```diff
@@ -93,18 +93,32 @@ where
     }
 }
 
-// A dynamic value when we don't know exactly what we are deserializing.
-// Useful for debugging.
+pub trait BencodeValueBufConstraint: std::hash::Hash + Eq + ByteBufT {}
```

Comment:
I don't like this - it's easier to depend on std::hash::Hash + Eq in implementation of ByteBufT, so you'll just have: "pub enum BencodeValue<BufT: ByteBufT>" if needed.

But at the same time I'm not sure why change the explicit bounds in the first place - it just increases the surface aread of the PR needlessly.